allennlp/modules/text_field_embedders/basic_text_field_embedder.py

```diff
@@ -81,8 +81,10 @@ def forward(self, text_field_input: Dict[str, torch.Tensor], num_wrapping_dims:
             embedded_representations.append(token_vectors)
         return torch.cat(embedded_representations, dim=-1)
 
+    # This is some unusual logic, it needs a custom from_params.
```

Comment:
Is it the logic, or just the `Dict[str, List[str]]`?  If we handled that `Dict` type, it looks like this would be standard, right?

Reply:
no, because the problem here is that the keys are not the names of function arguments (which is what the automatic `from_params` assumes), they're the names of token_indexers. 

that is, although the constructor takes `token_embedders: Dict[str, TokenEmbedder]`, the params object doesn't have a `token_embedders` key, the params object is itself the dict